subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -4056,12 +4056,12 @@ static inline void event_phy_upd_ind_prep(struct ll_conn *conn,
 			rx->hdr.link->mem = conn->llcp_rx;
 			conn->llcp_rx = rx;
 
-#if defined(CONFIG_BT_CTLR_DATA_LENGTH)
 			/* reserve rx node for DLE event generation */
-			rx = ll_pdu_rx_alloc();
-			rx->hdr.link->mem = conn->llcp_rx;
-			conn->llcp_rx = rx;
-#endif /* CONFIG_BT_CTLR_DATA_LENGTH */
+			if (IS_ENABLED(CONFIG_BT_CTLR_DATA_LENGTH)) {
+				rx = ll_pdu_rx_alloc();
```

Comment:
Is this missing an check for `rx == NULL`?

Reply:
Allocation is checked in lines 4003-4010: https://github.com/zephyrproject-rtos/zephyr/pull/36515/files#diff-135b49c1c599aa1060145905101574cb58224b39926a1e1a04dda6ff6e8702a0R4003-R4010

Reply:
So the allocation is reserved in those lines and checked there? Interesting. 

Is there a reason why we are not pre-allocating the PDU from line 4095 there as well then?

Reply:
`rx = ll_pdu_rx_alloc_peek(2);` 

The peek takes the buffer count to ensure the requested number of buffers are available in the free pool. The buffer count of `2` ensure, both PHY update and Data Length update notifications will get the buffer allocated. Hence, both allocations at line 4061 and 4095 are pre-checked.

Reply:
Alright, but shouldn't we remove the check in line 4096 then?

Reply:
sorry, was confused with the hidden lines of codes.

The peek is validation for lines 4055 and 4061, which is the PHY Update IND requested to be initiated, two buffers get allocated and reserved for future use.

For line 4095, there is no pre-check, and by design, only one buffer has been reserved when being in peripheral role, which is the received PDU buffer. Hence, an attempt to allocate the second buffer in line 4095 has a check.

Reply:
Is there a reason why we don't pre-check the allocation in line 4095? Should/can we?

Reply:
We do not need pre-check as line 4095 is the allocation and first thing being done in the `else if` at line 4083 (which is when the radio event is at the instant)